nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrProcessor.java

```diff
@@ -176,70 +172,11 @@ protected final KeytabUser getKerberosKeytabUser() {
-                if (solrLocation.startsWith("https:") && sslContextService == null) {
-                    problems.add(new ValidationResult.Builder()
-                            .subject(SSL_CONTEXT_SERVICE.getDisplayName())
-                            .valid(false)
-                            .explanation("an SSLContextService must be provided when using https")
-                            .build());
-                } else if (solrLocation.startsWith("http:") && sslContextService != null) {
-                    problems.add(new ValidationResult.Builder()
-                            .subject(SSL_CONTEXT_SERVICE.getDisplayName())
-                            .valid(false)
-                            .explanation("an SSLContextService can not be provided when using http")
-                            .build());
-                }
-            }
-        }
-
-        // Validate that we username and password are provided together, or that neither are provided
-        final String username = context.getProperty(BASIC_USERNAME).evaluateAttributeExpressions().getValue();
-        final String password = context.getProperty(BASIC_PASSWORD).evaluateAttributeExpressions().getValue();
-
-        final boolean basicUsernameProvided = !StringUtils.isBlank(username);
-        final boolean basicPasswordProvided = !StringUtils.isBlank(password);
-
-        if (basicUsernameProvided && !basicPasswordProvided) {
-            problems.add(new ValidationResult.Builder()
-                    .subject(BASIC_PASSWORD.getDisplayName())
-                    .valid(false)
-                    .explanation("a password must be provided for the given username")
-                    .build());
-        }
-
-        if (basicPasswordProvided && !basicUsernameProvided) {
-            problems.add(new ValidationResult.Builder()
-                    .subject(BASIC_USERNAME.getDisplayName())
-                    .valid(false)
-                    .explanation("a username must be provided for the given password")
-                    .build());
-        }
-
-        // Validate that only kerberos or basic auth can be set, but not both
-        final KerberosCredentialsService kerberosCredentialsService = context.getProperty(KERBEROS_CREDENTIALS_SERVICE).asControllerService(KerberosCredentialsService.class);
-        if (kerberosCredentialsService != null && basicUsernameProvided && basicPasswordProvided) {
-            problems.add(new ValidationResult.Builder()
-                    .subject(KERBEROS_CREDENTIALS_SERVICE.getDisplayName())
-                    .valid(false)
-                    .explanation("basic auth and kerberos cannot be configured at the same time")
-                    .build());
+        List<ValidationResult> _temp = new ArrayList<>(validateConnectionDetails(context));
+        if (_temp.size() == 0 && context.getProperty(CLIENT_SERVICE).isSet()) {
```

Comment:
The onScheduled method looks like it hasn't been changed and is not checking to see if a client service is present, and is still creating its own client.

We will also need to update the onStopped method to only close the client when it didn't come from the service.